src/components/Workspace.jsx

```diff
@@ -140,10 +144,12 @@ class Workspace extends React.Component {
   }
 
   _handleComponentMinimized(componentName) {
+    this.props.dispatch(editorsResetFlex());
```

Comment:
This is a pretty random/specific thing to throw in when we’re having a high-level discussion, BUT since we’re talking about pushing things down into the reducers, I had this on the brain. One of the design principles underlying my in-progress refactor of actions/reducers/sagas is that actions should communicate _what happened_, and the question of _what to do_ should be left to reducers and sagas. So, in this particular case, I don’t think we want an `editorsResetFlex` action—instead, the UI reducer should consume `MINIMIZE_COMPONENT`/`MAXIMIZE_COMPONENT` and reset the flex in response to those actions.